### PR TITLE
fix: Build absolute filePath so problems tab links to error location correctly

### PR DESCRIPTION
**Pull request by klewis-sfdc**
### What does this PR do?

A follow up to PR #4241 to fix the fileUri when constructed after a deployment error.

### What issues does this PR fix or reference?
##4228, @W-11350093@

### CODE REVIEW

File: packages/salesforcedx-vscode-core/src/diagnostics/diagnostics.ts [hunk 1]
```diff
@@ -143,3 +144,16 @@ export function handleDeployDiagnostics(
 
   return errorCollection;
 }
+
+export function getAbsoluteFilePath(
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-core/src/diagnostics/diagnostics.ts:148]: Should we put this in the same place as where getRootWorkspacePath lives?

File: packages/salesforcedx-vscode-core/src/diagnostics/diagnostics.ts [hunk 2]
```diff
@@ -143,3 +144,16 @@ export function handleDeployDiagnostics(
...
+  filePath: string | undefined,
+  workspacePath: string
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-core/src/diagnostics/diagnostics.ts:150]: should this default to the root?

File: packages/salesforcedx-vscode-core/test/vscode-integration/diagnostics/index.test.ts [hunk 3]
```diff
@@ -246,4 +247,22 @@ describe('Diagnostics', () => {
...
+    const filePath = `${workspacePath}/src/classes/Testing.cls`;
+    const asoluteFilePath = getAbsoluteFilePath(filePath, workspacePath);
+    expect(asoluteFilePath).to.equal(filePath);
+  });
+
+  it('Should use the workspace path as fileUri when filePath is undefined', () => {
+    const filePath = undefined;
+    const asoluteFilePath = getAbsoluteFilePath(filePath, workspacePath);
+    expect(asoluteFilePath).to.equal(workspacePath);
+  });
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-core/test/vscode-integration/diagnostics/index.test.ts:267]: I checked out the failing test and I'd just update it to the following: 
```
 const [fileDesc, reportedError] = setDiagnosticsStub.getCall(
            index
          ).args;
          const expectedFile = vscode.Uri.file(
            getAbsoluteFilePath(row.filePath, getRootWorkspacePath())
          );
          expect(fileDesc).to.deep.equal(expectedFile);
          expect(reportedError).to.deep.equal([
            {
              message: row.error,
              range: new vscode.Range(
                row.lineNumber - 1,
                row.columnNumber - 1,
                row.lineNumber - 1,
                row.columnNumber - 1
              ),
              severity: vscode.DiagnosticSeverity.Error,
              source: row.type
            }
          ]);
``` lines 457-476

It separates the the assertions so we don't have to run it to know which on is failing.  A very uningration solution as it pulls the utility into the test, but it was the best path I could come up with.

**Review by gbockus-sf — APPROVED** [review]
Passed my QE: 
✅ Verified navigation from problem to file when file is open 
✅ Verified navigation from problem to file when file is closed
✅ Verified navigation from problem to file when file is open but not the active editor
✅ Verified when problem is generated during active editing 
✅ Verified navigation from problem when the problem is reported after deploy from navigation
✅ Verified navigation from problem to file deploy via Deploy this source to Org

:shipit:

File: packages/salesforcedx-vscode-core/test/vscode-integration/diagnostics/index.test.ts (hunk 3 above)
Comment (klewis-sfdc) [reply packages/salesforcedx-vscode-core/test/vscode-integration/diagnostics/index.test.ts:267]: I updated the var names in this method to more closely reflect the params of the method where they are defined.

**Pull request merged**